src/gpu/intel/jit/ngen/ngen_level_zero.hpp

```diff
@@ -36,6 +44,54 @@ class level_zero_error : public std::runtime_error {
     ze_result_t status;
 };
 
+// Dynamically loaded level_zero functions
+namespace {
+
+void *find_ze_symbol(const char *symbol) {
+#if defined(__linux__)
+    void *handle = dlopen("libze_loader.so.1", RTLD_NOW | RTLD_LOCAL);
+#elif defined(_WIN32)
+    // Use LOAD_LIBRARY_SEARCH_SYSTEM32 flag to avoid DLL hijacking issue.
+    HMODULE handle = LoadLibraryExA(
+            "ze_loader.dll", nullptr, LOAD_LIBRARY_SEARCH_SYSTEM32);
+#endif
+    if (!handle) throw level_zero_error{ZE_RESULT_ERROR_UNINITIALIZED};
+
+    using zeInit_decl_t = ze_result_t (*)(ze_init_flags_t flags);
+    const ze_init_flags_t default_ze_flags = 0;
+#if defined(__linux__)
+    static const ze_result_t ze_result = reinterpret_cast<zeInit_decl_t>(
```

Comment:
You can actually omit the calls to `zeInit` as the caller must invoke L0 prior to entering nGEN (to locate the device, create a context, etc.).